ledger/heavy/handler/handler.go

```diff
@@ -351,20 +359,48 @@ func (h *Handler) getCode(ctx context.Context, id *insolar.ID) (record.Code, err
 }
 
 func (h *Handler) handleHeavyPayload(ctx context.Context, genericMsg insolar.Parcel) (insolar.Reply, error) {
+	logger := inslogger.FromContext(ctx)
 	msg := genericMsg.Message().(*message.HeavyPayload)
 
 	storeRecords(ctx, h.RecordModifier, h.PCS, msg.PulseNum, msg.Records)
 	if err := storeIndexBuckets(ctx, h.IndexBucketModifier, msg.IndexBuckets, msg.PulseNum); err != nil {
 		return &reply.HeavyError{Message: err.Error(), JetID: msg.JetID, PulseNum: msg.PulseNum}, nil
 	}
-	if err := storeDrop(ctx, h.DropModifier, msg.Drop); err != nil {
-		return &reply.HeavyError{Message: err.Error(), JetID: msg.JetID, PulseNum: msg.PulseNum}, nil
+
+	{
+		pulse, err := h.PulseAccessor.Latest(ctx)
+		if err != nil {
+			return &reply.HeavyError{Message: err.Error(), JetID: msg.JetID, PulseNum: msg.PulseNum}, nil
+		}
+		futurePulse := pulse.NextPulseNumber
+		drop, err := storeDrop(ctx, h.DropModifier, msg.Drop)
+		switch {
```

Comment:
First two cases do `return` at the end., so `switch` is not necessary. We can flatten it.

Reply:
done